lib/vector-core/src/event/vrl_target.rs

```diff
@@ -497,27 +493,25 @@ fn target_get_metric<'a>(
 
     let value = value.get(path);
 
-    for paths in path.to_alternative_components(MAX_METRIC_PATH_DEPTH) {
-        match paths.as_slice() {
-            ["name"] | ["kind"] | ["type"] | ["tags", _] => return Ok(value),
-            ["namespace"] | ["timestamp"] | ["interval_ms"] | ["tags"] => {
-                if let Some(value) = value {
-                    return Ok(Some(value));
-                }
-            }
-            _ => {
-                return Err(MetricPathError::InvalidPath {
-                    path: &path.to_string(),
-                    expected: VALID_METRIC_PATHS_GET,
-                }
-                .to_string())
-            }
+    let Some(paths) = path.to_alternative_components(MAX_METRIC_PATH_DEPTH) else {
+        return Ok(None);
```

Comment:
Before it'd return `vec![]` on the effective None case and therefore the for loop would never run and would eventually return `Ok(None);`. This just makes it more explicit + the comment was very misleading